File: src/main/java/com/finances/dto/ExpenseFilterDTO.java
package com.finances.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.finances.entity.User;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ExpenseFilterDTO {

    Integer categoryId;
    LocalDateTime startDate;
    LocalDateTime endDate;
    String name;
    User user;

}

File: src/main/kotlin/com/finances/dto/ExpenseFilterDTO.kt
package com.finances.dto

import java.time.LocalDateTime

data class ExpenseFilterDTO(
    var categoryId: Int = 0,
    var startDate: LocalDateTime? = null,
    var endDate: LocalDateTime? = null,
    var name: String? = null,
    var userId: Int = 0
)
